pkg/controller/core.oam.dev/v1alpha2/application/suite_test.go

```diff
@@ -168,3 +175,72 @@ var _ = AfterSuite(func() {
 	Expect(err).ToNot(HaveOccurred())
 	close(stop)
 })
+
+type FakeRecorder struct {
+	Events  chan string
+	Message map[string][]*Events
+}
+
+type Events struct {
+	Name      string
+	Namespace string
+	EventType string
+	Reason    string
+	Message   string
+}
+
+func (f *FakeRecorder) Event(object runtime.Object, eventtype, reason, message string) {
+	if f.Events != nil {
+		objectMeta, err := meta.Accessor(object)
+		if err != nil {
+			return
+		}
+
+		event := &Events{
+			Name:      objectMeta.GetName(),
+			Namespace: objectMeta.GetNamespace(),
+			EventType: eventtype,
+			Reason:    reason,
+			Message:   message,
+		}
+		f.Events <- fmt.Sprintf("%s %s %s %s %s", objectMeta.GetName(), objectMeta.GetNamespace(), eventtype, reason, message)
```

Comment:
It's not necessary as you have stored all the information in `f.Message`.

Reply:
copy